### ISSUE TITLE
Definition was declared but never used in document when using YAML anchors

### ISSUE THREAD

**Issue by meeech** [swagger-api/swagger-editor#1285]
3.0.5
Created YAML Anchor to reference an ENUM, but then end up with warning Definition was declared... 
Not sure if this is known issue or not (may be acceptable to have this warning) - so, while strictly the definition isn't used in the document, the anchor which links to the definition is used in another definition. 
`NetworkInterfaceState:
  type: "string"
  enum: &NISTATE
    - "UP"
    - "DOWN"
SystemState:
  type: "object"
  properties:
    master:
      type: "boolean"
    networkInterfaces:
      type: "object"
      properties:
        em1: 
          type: "string"
          enum: *NISTATE
        em2: 
          type: "string"
          enum: *NISTATE`

**Comment by webron** [comment]
It's a validation issue we can change. However, I don't really see why you'd use an anchor for that specific case, if you only reference it from models. Just reference `em1` and `em2` to `NetworkInterfaceState` and that's it.

**Comment by meeech** [comment]
that works, except then I get an error sibling values are not allowed along side $ref (trying to use example: )

**Comment by webron** [comment]
Yeah, that should be a warning and not an error, but why would get that?
I'm saying you should have it like this:
```yaml
NetworkInterfaceState:
  type: "string"
  enum:
    - "UP"
    - "DOWN"
SystemState:
  type: "object"
  properties:
    master:
      type: "boolean"
    networkInterfaces:
      type: "object"
      properties:
        em1: 
          $ref: '#/definitions/NetworkInterfaceState'
        em2: 
          $ref: '#/definitions/NetworkInterfaceState'
```
That shouldn't yield that error.

**Comment by meeech** [comment]
Yes. But I mean, that was reduced example
I want to show in options in the example displayed, so for some I put the example: 'foo' to show one of the other options. 
so looks more like this:

```yaml
      networkInterfaces:
        type: "object"
        description: "Note that the returned keys may vary depending on system setup (eg: VM vs physical hardware)"
        properties:
          em1: 
            $ref: "#/definitions/NetworkInterfaceState"
          em2: 
            $ref: "#/definitions/NetworkInterfaceState"
          p15p1: 
            type: "string"
            enum: *NISTATE
            example: "DOWN"
```

**Comment by webron** [comment]
If you want to set a different example for each instance, then you're right, the $ref won't work.

Why you'd want to have an example for a field that has two possible enum values... that's a different issue ;)

**Comment by meeech** [comment]
well, its for client facing doc :D so I like to make examples explicit when possible, and there are other enums with more options (this was just one example)

**Comment by hkosova** [comment]
@meeech A possible workaround to avoid the warning is to move the definitions referenced via YAML anchors to an arbitrary section prefixed with `x-`, say, `x-types`. The unused definition detector ignores `x-` keys.

```
x-types:
  NetworkInterfaceState:
    type: "string"
    enum: &NISTATE
      - "UP"
      - "DOWN"
```
But in this case these definitions won't appear in the Models section.

**Comment by meeech** [comment]
@hkosova thanks. good to know

**Comment by MichakrawSB** [comment]
We’ve released Swagger Editor v5! 🎉 Check out the details here: [Inside the New Swagger Editor](https://swagger.io/blog/inside-the-new-swagger-editor/).
We’re closing old issues related to previous versions. If you think any of them are still relevant, please open a new issue – this helps us prioritize what matters most. Thanks for your activity! 🚀
